tools/benchmark/src/mocha/memoryTestRunner.ts

```diff
@@ -337,54 +298,102 @@ export function benchmarkMemory(testObject: IMemoryTestObject): Test {
+				).after.memoryUsage.push(process.memoryUsage());
+				(benchmarkStats.customData["Samples used"] as MemorySampleData).after.heap.push(
+					v8.getHeapStatistics(),
+				);
+				(
+					benchmarkStats.customData["Samples used"] as MemorySampleData
+				).after.heapSpace.push(v8.getHeapSpaceStatistics());
+
+				benchmarkStats.customData.Iterations =
+					(benchmarkStats.customData.Iterations as number) + 1;
 
 				const heapUsedArray: number[] = [];
-				for (let i = 0; i < benchmarkStats.samples.before.memoryUsage.length; i++) {
+				for (
+					let i = 0;
+					i <
+					(benchmarkStats.customData["Samples used"] as MemorySampleData).before
+						.memoryUsage.length;
+					i++
+				) {
 					heapUsedArray.push(
-						benchmarkStats.samples.after.memoryUsage[i].heapUsed -
-							benchmarkStats.samples.before.memoryUsage[i].heapUsed,
+						(benchmarkStats.customData["Samples used"] as MemorySampleData).after
+							.memoryUsage[i].heapUsed -
+							(benchmarkStats.customData["Samples used"] as MemorySampleData).before
+								.memoryUsage[i].heapUsed,
 					);
 				}
 				heapUsedStats = getArrayStatistics(heapUsedArray, options.samplePercentageToUse);
 
 				// Break if max elapsed time passed, only if we've reached the min sample count
 				if (
-					benchmarkStats.runs >= options.minSampleCount &&
+					(benchmarkStats.customData.Iterations as number) >= options.minSampleCount &&
 					timer.toSeconds(startTime, timer.now()) > options.maxBenchmarkDurationSeconds
 				) {
 					break;
 				}
 			} while (
-				benchmarkStats.runs < options.minSampleCount ||
+				(benchmarkStats.customData.Iterations as number) < options.minSampleCount ||
 				heapUsedStats.marginOfErrorPercent > options.maxRelativeMarginOfError
 			);
-			benchmarkStats.stats = heapUsedStats;
+
+			benchmarkStats.customDataFormatters.Iteration = (value): string =>
+				(value as number).toString();
+
+			benchmarkStats.customData["Heap Used Avg"] = heapUsedStats.arithmeticMean;
+			benchmarkStats.customDataFormatters["Heap Used Avg"] = (value): string =>
+				prettyNumber(value as number, 2);
+
+			benchmarkStats.customData["Margin of Error"] = heapUsedStats.marginOfError;
+			benchmarkStats.customDataFormatters["Margin of Error"] = (value): string =>
+				`±${prettyNumber(value as number, 2)}`;
+
+			benchmarkStats.customData["Relative Margin of Error"] =
+				heapUsedStats.marginOfErrorPercent;
+			benchmarkStats.customDataFormatters["Relative Margin of Error"] = (value): string =>
+				`±${prettyNumber(value as number, 2)}`;
+
+			// benchmarkStats.customData["samples"] = heapUsedStats.samples;
+
+			benchmarkStats.customData["Heap Used StdDev"] = heapUsedStats.standardDeviation;
+			benchmarkStats.customDataFormatters["Heap Used StdDev"] = (value): string =>
+				prettyNumber(value as number, 2);
+
+			// benchmarkStats.customData["standardErrorOfMean"] = heapUsedStats.standardErrorOfMean;
+			// benchmarkStats.customData["variance"] = heapUsedStats.variance;
```

Comment:
We still want these. Pending to uncomment and provide formatters?

Reply:
Synced offline. Should covered this one.